pkg/configurator/validating_webhook.go

```diff
@@ -0,0 +1,346 @@
+package configurator
+
+import (
+	"context"
+	"crypto/tls"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"regexp"
+	"strconv"
+	"time"
+
+	"k8s.io/api/admission/v1beta1"
+	admissionv1beta1 "k8s.io/api/admissionregistration/v1beta1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/serializer"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/kubernetes"
+
+	"github.com/openservicemesh/osm/pkg/certificate"
+	"github.com/openservicemesh/osm/pkg/constants"
+	"github.com/openservicemesh/osm/pkg/webhook"
+)
+
+var (
+	codecs       = serializer.NewCodecFactory(runtime.NewScheme())
+	deserializer = codecs.UniversalDeserializer()
+
+	//boolFieldsInConfigMap are the fields in osm-config that take in a boolean
+	boolFieldsInConfigMap = []string{"egress", "enable_debug_server", "permissive_traffic_policy_mode", "prometheus_scraping", "tracing_enable", "use_https_ingress"}
+
+	//ValidEnvoyLogLevels is a list of envoy log levels
+	ValidEnvoyLogLevels = []string{"trace", "debug", "info", "warning", "warn", "error", "critical", "off"}
+)
+
+const (
+	// ValidatingWebhookName is the name of the validating webhook used for validating osm-config
+	ValidatingWebhookName = "validating-webhook.k8s.io"
+
+	//webhookUpdateConfigMapis the HTTP path at which the webhook expects to receive configmap update events
+	webhookUpdateConfigMap = "/validate-webhook"
+	listenPort             = 9090
+
+	//MustBeBool is the reason for denial for a boolean field
+	MustBeBool = ": must be a boolean"
+
+	//MustBeValidLogLvl is the reason for denial for envoy_log_level field
+	MustBeValidLogLvl = ": invalid log level"
+
+	//MustBeValidTime is the reason for denial for incorrect syntax for service_cert_validity_duration field
+	MustBeValidTime = ": invalid time format must be a sequence of decimal numbers each with optional fraction and a unit suffix"
+
+	//MustBeLessThanAYear is the reason for denial for service_cert_validity_duration field
+	MustBeLessThanAYear = ": must be max 8760H (1 year)"
+
+	//MustFollowSyntax is the reason for denial for tracing_address field
+	MustFollowSyntax = ": invalid hostname syntax"
+
+	//MustbeInt is the reason for denial for incorrect syntax for tracing_port field
+	MustbeInt = ": must be an integer"
+
+	//MustBeInPortRange is the reason for denial for tracing_port field
+	MustBeInPortRange = ": must be between 0 and 65535"
+
+	//CannotChangeMetadata is the reason for denial for changes to configmap metadata
+	CannotChangeMetadata = ": cannot change metadata"
```

Comment:
Most of these don't need to be exported, could you use unexported constants where applicable?

Reply:
I have them as exported for the unit test for the `validate` function

Reply:
@shalier which test?

Reply:
@shashankram it's in validating_webhook_test.go at L246 and L260 for `TestValidate`

Reply:
But the test is within the same package, so you don't need to export it?

Reply:
oops, you're right, don't know what i was doing there 🤦